woocommerce/woocommerce
# Show a warning modal when user tries to leave the assember hub

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #41494 .

This PR shows a warning modal when the user tries to leave the assember hub page by clicking the browser's back button.

One notable change is the removal of popstate and pushState event handlers (https://github.com/woocommerce/woocommerce/pull/41596/commits/6c9db0551eab4958f59513b512a3f2a09b9c769c ) from the iframe. They were interrupting the history and I could not make the back button work as expected without removing them. As far as I can see, everything works as expected, but it should be double-checked. @ilyasfoo ✋ Could you confirm this change please? 






### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Make sure you are testing this with a Jetpack connected instance of WooCommerce, as the AI text completion endpoint can only be called with a Jetpack connected account.

1. Create a new JN site for this branch with WooCommerce Beta
2. Enable `customize-store` features flag from `Tools -> WCA Test Helper`
3. Go to `WooCommerce -> Home` and click on `Customize your store` task
4. Proceed to the assember hub page.
5. Click around a few items.
6. Click back button from the assember hub main page.
7. The warning modal should render.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Show a warning modal when the user tries to leave the assember hub (CYS).

#### Comment 

</details>
